app/src/main/kotlin/com/intive/picover/profile/viewmodel/ProfileViewModel.kt

```diff
@@ -35,7 +36,17 @@ class ProfileViewModel @Inject constructor(
 		}
 	}
 
+	fun updateAvatar(uri: Uri) {
+		viewModelScope.launch {
+			_profile.value = Loading
+			authRepository.updateUserAvatar(uri)
+			fetchProfile()
```

Comment:
Aren't we loosing Loading state now, if the update is finished ?

Reply:
What do you mean? I set the profile in the loading state to show the progress bar and when the avatar will be updating and it will finish correctly. We run fetchProfile() to refresh data.